inventory_requests/business_logic/loan_logic.py

```diff
@@ -4,13 +4,16 @@
 
 
 def return_loan_logic(loan, quantity=None):
-    if loan.returned_timestamp is None and loan.loaned_timestamp is not None:
-        if quantity is None or quantity == loan.quantity:
+    if loan.returned_timestamp is None and loan.loaned_timestamp is not None and loan.returned_quantity < loan.quantity:
+        if quantity is None or loan.returned_quantity + quantity == loan.quantity:

```

Comment:
wouldn't this update the returned timestamp even if the loan is only partially returned? or is that intended behavior?